### DISCUSSION TITLE
Query Process by correlated variables

### DISCUSSION THREAD

**Discussion by MichaelVonB** [bpm-crafters/bpm-crafters#108] [Ideas]
### Scenario

* Library version: 0.3.0
* Description of your use case: I want to find a process and/or its variables by some correlated variables (not the execution Id) to check where the current state. Depending on the information certain actions might be possible or not. 

### Current Behaviour
No query api present

### Wanted Behaviour
Provide a way to query active processes/variables by some correlated variables

### Possible Workarounds
n.a.

**Comment by zambrovski** [comment]
Providing a query API exposes a lot of internals of the process engine. For example the pure presence of internal process variables is a strong assumption. During the design of the Process Engine API, we are carefully selecting provided API in order not to create a too strong dependency to the process engine.
